hotsos/defs/scenarios/mysql/bugs/lp2039444.yaml

```diff
@@ -0,0 +1,16 @@
+checks:
+  has_2039444:
+    input:
+      path: 'var/log/juju/*mysql*.log'
+    start: '([\d-]+) ([\d:]+) .+ Failed adding instance .+ to cluster'
+    end: '[\s\S]+mysqlsh\.DBError: MySQL Error \(\d+\): Cluster\.add_instance: Could not open connection to .+: Host .+ is not allowed to connect to this MySQL server'
```

Comment:
I reckon the `[\s\S]+` is redundant, and that might be the reason why the CI fails. You can re-run the failing test locally by running `tox -e py3 -- test_bugs_lp2039444`

Reply:
yes that's correct, the mysqlsh line in your test has nothing before "mysqlsh..." hence why that does not match. I have also landed support for sequence searches in scenario checks so please rebase your patch. Those two things should result in a a pass (i have tested it and it works).

```
diff --git a/hotsos/defs/scenarios/mysql/bugs/lp2039444.yaml b/hotsos/defs/scenarios/mysql/bugs/lp2039444.yaml
index a59b7065..68bcdf12 100644
--- a/hotsos/defs/scenarios/mysql/bugs/lp2039444.yaml
+++ b/hotsos/defs/scenarios/mysql/bugs/lp2039444.yaml
@@ -3,7 +3,7 @@ checks:
     input:
       path: 'var/log/juju/*mysql*.log'
     start: '([\d-]+) ([\d:]+) .+ Failed adding instance .+ to cluster'
-    end: '[\s\S]+mysqlsh\.DBError: MySQL Error \(\d+\): Cluster\.add_instance: Could not open connection to .+: Host .+ is not allowed to connect to this MySQL server'
+    end: 'mysqlsh\.DBError: MySQL Error \(\d+\): Cluster\.add_instance: Could not open connection to .+: Host .+ is not allowed to connect to this MySQL server'
 conclusions:
   lp2039444:
     decision: has_2039444
```

Reply:
rebased and addressed Ed's review feedback. @mustafakemalgilor : Ed had to improve hotsos to support multi-line regex search. My attempt with [\s\S]+ would work with native python regex.